### PR TITLE
Add functional test

### PR DESCRIPTION
**Pull request by linusnorton**
Add a functional test to trigger the notification

### CODE REVIEW

File: src/functionalTest/resources/request/casedata/ccd-callback-casedata-payment-made.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
+  "divorceOrDissolution": "divorce",
+  "inferredPetitionerGender": "female",
+  "inferredRespondentGender": "male",
+  "marriageDate": "2020-04-28",
+  "petitionerFirstName": "Billy",
+  "petitionerMiddleName": "B",
+  "petitionerLastName": "Bob",
+  "petitionerContactDetailsConfidential": "keep",
+  "petitionerEmail": "linus.norton@solirius.com",
```
Comment (nitinprabhu) [inline src/functionalTest/resources/request/casedata/ccd-callback-casedata-payment-made.json:10]: worth using dummy email ?

File: src/functionalTest/java/uk/gov/hmcts/divorce/citizen/PaymentMadeTest.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+@SpringBootTest
+public class PaymentMadeTest extends FunctionalTestSuite {
+
+    private static final String REQUEST = "classpath:request/casedata/ccd-callback-casedata-payment-made.json";
+    private static final String RESPONSE = "classpath:responses/ccd-callback-casedata-payment-made.json";
+
+    @Test
+    public void shouldPassValidationAndGiveSuccessWhenCaseDataValid() throws IOException {
+        REQUEST.replace("2020-04-28", LocalDate.now().minus(1, YEARS).minus(1, DAYS).toString());
```
Comment (nitinprabhu) [inline src/functionalTest/java/uk/gov/hmcts/divorce/citizen/PaymentMadeTest.java:38]: date can be a constant

File: src/main/java/uk/gov/hmcts/divorce/common/model/CaseData.java [hunk 3]
```diff
@@ -722,6 +722,7 @@
         label = "Date of submission",
         access = {DefaultAccess.class}
     )
+    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss.SSS")
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/divorce/common/model/CaseData.java:725]: do we need milliseconds ?

Comment (linusnorton) [reply src/main/java/uk/gov/hmcts/divorce/common/model/CaseData.java:725]: It's what CCD expects: https://tools.hmcts.net/confluence/pages/viewpage.action?spaceKey=RCCD&title=CCD+Supported+Field+Types&__ncforminfo=mand5-SxRtTmsp4kjkR-uw0VanXD-JnuZMs1BZeym6mK0V1-26DCB4-gvqUWF2y0iA6bKT1xLf13fhTyzs_nmw==#CCDSupportedFieldTypes-ccdTypeDateTime

File: src/functionalTest/resources/request/casedata/ccd-callback-casedata-payment-made.json (hunk 1 above)
Comment (linusnorton) [reply src/functionalTest/resources/request/casedata/ccd-callback-casedata-payment-made.json]: yes, forgot to update after testing :D

File: src/functionalTest/java/uk/gov/hmcts/divorce/citizen/PaymentMadeTest.java (hunk 2 above)
Comment (linusnorton) [reply src/functionalTest/java/uk/gov/hmcts/divorce/citizen/PaymentMadeTest.java:38]: I'm not sure it adds much value TBH

**Review by nitinprabhu — APPROVED** [review]

File: src/functionalTest/java/uk/gov/hmcts/divorce/citizen/PaymentMadeTest.java (hunk 2 above)
Comment (nitinprabhu) [reply src/functionalTest/java/uk/gov/hmcts/divorce/citizen/PaymentMadeTest.java:38]: not strong about it think should be fine the way it is

**Pull request merged**